src/structures/Guild.js

```diff
@@ -392,19 +392,24 @@ class Guild extends Base {
    */
 
   /**
-   * Fetches a collection of banned users in this guild.
-   * @returns {Promise<Collection<Snowflake, BanInfo>>}
-   */
-  fetchBans() {
-    return this.client.api.guilds(this.id).bans.get().then(bans =>
-      bans.reduce((collection, ban) => {
-        collection.set(ban.user.id, {
-          reason: ban.reason,
-          user: this.client.users.add(ban.user),
-        });
-        return collection;
-      }, new Collection())
-    );
+   * Fetches a single banned user or a collection of banned users in this guild.
+   * @param {string} [id] Optional id of a banned user.
+   * @returns {Promise<Collection<Snowflake, BanInfo> | BanInfo>}
+   */
+  fetchBans(id) {
+    return this.client.api.guilds(this.id).bans(id).get()
+      .then(data => {
+        if (id) {
+          return { reason: data.reason, user: this.client.users.add(data.user) };
+        }
+        return data.reduce((collection, ban) => {
+          collection.set(ban.user.id, {
```

Comment:
You could make this a single statement arrow function, since `Map#set` returns the map. 👀